service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusServiceUnitTest.java

```diff
@@ -0,0 +1,129 @@
+                                                                       ERROR_MESSAGE,
+                                                                       ehrExtractStatusServiceSpy.getClass().getSimpleName());
+    }
+
+    @Test
+    void ehrExtractStatusThrowsExceptionWhenConversationIdNotFound() {
+        EhrExtractStatusService ehrExtractStatusServiceSpy = spy(ehrExtractStatusService);
+        String conversationId = "11111";
+        Instant currentInstant = Instant.now();
+        Instant nineDaysAgo = currentInstant.minus(Duration.ofDays(NINE_DAYS));
+        Optional<EhrExtractStatus> ehrExtractStatus = Optional.of(EhrExtractStatus.builder().updatedAt(nineDaysAgo).build());
+
+        doReturn(true).when(ehrExtractStatusServiceSpy).isEhrStatusWaitingForFinalAck(conversationId);
+        doReturn(false).when(ehrExtractStatusServiceSpy).hasFinalAckBeenReceived(conversationId);
+        doReturn(Optional.empty()).when(ehrExtractStatusRepository).findByConversationId(conversationId);
+        when(ack.getErrors()).thenReturn(null);
+        when(ack.getReceived()).thenReturn(currentInstant);
+
+        Exception exception = assertThrows(NonExistingInteractionIdException.class, () ->
+            ehrExtractStatusServiceSpy.updateEhrExtractStatusAck(conversationId, ack));
+
+        assertEquals("Received a ACK message that is not recognised with conversation_id: " + conversationId,
+                     exception.getMessage());
+
+        verify(ehrExtractStatusServiceSpy, never()).updateEhrExtractStatusError(conversationId,
+                                                                       ERROR_CODE,
+                                                                       ERROR_MESSAGE,
+                                                                       ehrExtractStatusServiceSpy.getClass().getSimpleName());
+    }
+
+    @Test
+    void ehrExtractStatusIsUpdatedWhenDelayBewteenAckUpdatesIsLessThan8Days() {
+        EhrExtractStatusService ehrExtractStatusServiceSpy = spy(ehrExtractStatusService);
+        String conversationId = "11111";
+        Instant currentInstant = Instant.now();
+        Instant eightDaysAgo = currentInstant.minus(Duration.ofDays(EIGHT_DAYS));
+        Optional<EhrExtractStatus> ehrExtractStatus = Optional.of(EhrExtractStatus.builder().updatedAt(eightDaysAgo).build());
+
+        doReturn(true).when(ehrExtractStatusServiceSpy).isEhrStatusWaitingForFinalAck(conversationId);
+        doReturn(false).when(ehrExtractStatusServiceSpy).hasFinalAckBeenReceived(conversationId);
+        doReturn(ehrExtractStatus).when(ehrExtractStatusRepository).findByConversationId(conversationId);
+        when(ack.getErrors()).thenReturn(null);
+        when(ack.getReceived()).thenReturn(currentInstant);
+        doReturn(ehrExtractStatus.get())
+            .when(mongoTemplate).findAndModify(any(Query.class), any(Update.class), any(), any(Class.class));
```

Comment:
Is there any way to make the `any()` more restrictive? 😄 

```suggestion
            .when(mongoTemplate).findAndModify(any(Query.class), any(Update.class), any(), any(Class.class));
```